src/lib/parser.rs

```diff
@@ -108,8 +108,51 @@ impl Parser {
     Ok(parser)
   }
   pub fn next_char(&mut self) -> Option<char> {
+    // get next
     let letter = self.contents.get(self.index)?;
```

Comment:
Maybe this can be changed to:
```suggestion
    let letter = *self.contents.get(self.index)? as char;
```
This way you don't need to define the u8 chars under here and that cleanup the code and makes it much better readable because you can directly compare it to the char